app/Http/Controllers/ProductController.php

```diff
@@ -13,36 +14,50 @@ class ProductController extends Controller
     public function index()
     {
         $products = Product::select('id', 'name', 'brand_id', 'color_id', 'size_id', 'status')->paginate(10);
-          
+
         return view('admin.pages.product.index', compact('products'));
     }
+
     public function add()
     {
         $brands = Brand::select('id', 'name')->get();
         $colors = Color::select('id', 'name')->get();
-        $sizes = Size::select('id', 'name')->get();    
-        return view('admin.pages.product.form', compact('brands', 'colors', 'sizes'));
-    }
+        $sizes = Size::select('id', 'name')->get();
+        $categories = Category::select('id', 'name')->get();
+
+        return view('admin.pages.product.form', compact('brands', 'colors', 'sizes', 'categories'));
+    }  
+
     public function store(ProductValidation $request)
     {
-        Product::create($request->validated());
+        $product = Product::create($request->validated());
+        $product->attachCategories($request->validated(['category_ids']));
```

Comment:
create variable for $request->validated()
why use again attachCategories from the product? I already said to make private method see link:
https://github.com/jayesh-parmar/inventory-management/pull/25#discussion_r1342357701